File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-3.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int l = 0;
        unordered_map<char, int> mp;
        int res = 0;
        int mf = 0;
        for (int r = 0; r < s.size(); r++) {
            mp[s[r]]++;
            mf = max(mf, mp[s[r]]);
            while (r - l + 1 + - mf > k) {
                mp[s[l]]--;
                l++;
            }
            res = max(res, r - l + 1);
        } 
        return res;
    }
};

File: Data Structures & Algorithms/maximum-subarray/submission-3.java
class Solution {
    public int maxSubArray(int[] nums) {
        int res = nums[0];
        int cur_sum = 0;
        for (int num : nums) {
            cur_sum = Math.max(cur_sum + num, num);
            res = Math.max(cur_sum, res);
        }
        return res;
    }
}
